src/drivers/net/e1000.c

```diff
@@ -197,16 +201,17 @@ static void e1000_rx(struct net_device *dev) {
 			skb = rx_skbs[cur];
 			assert (skb);
 
-			rx_skbs[cur]= skb_alloc(ETH_FRAME_LEN);
+			rx_skbs[cur] = skb_alloc(E1000_MAX_RX_LEN);
 			if (rx_skbs[cur] == NULL)
 				panic("Can't allocate rx_skbs %d", cur);
 			rx_descs[cur].buffer_address = (uint32_t) rx_skbs[cur]->mac.raw;
 
-
-			skb->dev = dev;
-			netif_rx(skb);
-
-
+			len = rx_descs[cur].length - E1000_RX_CHECKSUM_LEN;
+			skb = skb_realloc(len, skb);
+			if (skb == NULL) {
+				skb->dev = dev;
```

Comment:
I guess, you already hate me, but this is a NULL pointer dereference. :fearful: 

How about this?
```c
if (!skb)
    goto drop_pack;
```